src/main/java/seedu/address/logic/commands/student/EditStudentCommand.java

```diff
@@ -71,25 +70,21 @@ private static Student createdEditedStudent(Student studentToEdit, EditStudentDe
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Person> lastShownList = model.getFilteredPersonList();
+        List<Student> lastShownList = model.getFilteredStudentList();
 
         if (index.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
         }
 
-        Person personToEdit = lastShownList.get(index.getZeroBased());
-        if (!(personToEdit instanceof Student)) {
-            throw new CommandException("The person you are trying to edit is not a student!");
-        }
-        Student studentToEdit = (Student) personToEdit;
+        Student studentToEdit = lastShownList.get(index.getZeroBased());
         Student editedStudent = createdEditedStudent(studentToEdit, editStudentDescriptor);
 
-        if (!studentToEdit.isSamePerson(editedStudent) && model.hasPerson(editedStudent)) {
+        if (!studentToEdit.isSamePerson(editedStudent) && model.hasStudent(editedStudent)) {
             throw new CommandException(MESSAGE_DUPLICATE_STUDENT);
         }
 
-        model.setPerson(studentToEdit, editedStudent);
-        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        model.setStudent(studentToEdit, editedStudent);
+        model.updateFilteredStudentList(Model.PREDICATE_SHOW_ALL_STUDENTS);
```

Comment:
this means the UI will only show the student list after an edit command? will it also show teacher list?

Reply:
At first, the `filteredStudentList` is initialised to all students. If the user filters the list and then does the `editStudent` command, this line of code will trigger the UI to relist all the students. The teacher list will not be affected.